runtime/autoload/provider/pythonx.vim

```diff
@@ -43,15 +54,17 @@ function! s:check_version(prog, ver, skip) abort
         \ ' -c "import sys; sys.stdout.write(str(sys.version_info[0]) + '.
         \ '\".\" + str(sys.version_info[1]))"'
   let min_version = (a:ver == 2) ? '2.6' : '3.3'
-  if system(a:prog . get_version) >= min_version
+  let prog_version = system(a:prog . get_version)
+  if prog_version =~ '^' . a:ver && prog_version >= min_version
```

Comment:
This is the real change in this PR: check that the major version of Python is the requested one. Before, Python 3 was accepted as a Python 2 interpreter (since 3.x >= 2).